pkg/kademlia/queue.go

```diff
@@ -4,57 +4,108 @@
 package kademlia
 
 import (
+	"container/heap"
 	"math/big"
 
+	"storj.io/storj/pkg/dht"
 	"storj.io/storj/pkg/pb"
 )
 
-// An Item is something we manage in a priority queue.
-type Item struct {
+//We've encapsulated our priorityQueue{} implementation (below) so that you don't
+//have to know how the Golang heap API works (nor our priority XOR logic)
+
+//XorQueue is a priority queue where the priority is key XOR distance
+type XorQueue struct {
+	pq *priorityQueue
+}
+
+//NewXorQueue returns a priorityQueue with priority based on XOR from targetBytes
+func NewXorQueue(nodes []*pb.Node, target dht.NodeID) XorQueue {
+	targetBytes := new(big.Int).SetBytes(target.Bytes())
+	pq := make(priorityQueue, len(nodes))
+	for i, node := range nodes {
+		pq[i] = &item{value: node, index: i}
+		pq[i].priority = new(big.Int).Xor(targetBytes, new(big.Int).SetBytes([]byte(node.GetId())))
```

Comment:
Let's not use big.Int for this. Create a `nodeID [32]byte` and use that throughout. Can be a separate PR.

Reply:
We currently derive our big.Int from dht.NodeId, which isn't fixed length.  We could use a MD5 to get a [16]byte or SHA-256 to get a [32]byte.  Thoughts on using a hash vs changing dht.NodeId?

Reply:
BigInt also provides a built in Xor method that [32]byte doesn't

Reply:
I'm also not sure why [32]byte is preferable to BigInt

Reply:
I think it would end up more readable than using `big.Int`. As an example that exact thing could be:

```
pq[i].priority = target.Xor(nodeID(node))
```